remote_attestation/rust/src/tests/crypto.rs

```diff
@@ -0,0 +1,217 @@
+//
+// Copyright 2021 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
```

Comment:
Have you considered moving this to a tests directory? https://doc.rust-lang.org/book/ch11-03-test-organization.html#the-tests-directory

Reply:
IIUC global `tests` directory is mostly used for integration tests, while `src/tests` is for unit tests.
Also I have several private functions that I test (`key_derivation_function`, `derive_session_keys`).
I mark them as `pub(crate)` so that they will be visible in tests. It's possible to include tests alongside `crypto.rs`, but there will be too much code in the same place.